### ISSUE TITLE
Generic Database connector client (8.7 Beta)

### ISSUE THREAD

**Issue by danajuratoni** [elastic/connectors-python#404]
## New Generic Database connector
New connector client to be released with 8.7 in Beta.
**8.7 Feature Freeze** on 7th Feb 2023
**8.7 GA** on Tue, Mar 28 2023

### Acceptance criteria
- [x] TDD in [GDrive](https://drive.google.com/drive/folders/1DN2y4TRXIuybX8vawHljkWozoyPEOYaf) for PostgreSQL, Oracle, MSSQL
- [x] QA plan in [GDrive](https://drive.google.com/drive/folders/1DN2y4TRXIuybX8vawHljkWozoyPEOYaf) for PostgreSQL, Oracle, MSSQL
- [x] Code merged in main following the defined PR process
     - [x] MS SQL https://github.com/elastic/connectors-python/pull/509
- [x] Functional tests performed
     - [x] MS SQL
- [x] Performance test/ reports against the last stable release branch of the connector framework and in Elastic Cloud 
- [x] Connector client documentation in repo for PostgreSQL, Oracle
     - [x] MS SQL
- [x] Demo of the connector in latest builds in Elastic Cloud
     - https://drive.google.com/drive/folders/1Uykjm51XdfxmhC3mxcNy8eAvMEaIOQhD
     - [x] MS SQL     
- [x] Ingestion engineer added a comment to this ticket to sign off - for PostgreSQL, Oracle, MSSQL
     - [x] Code backported to the correct version 
     - [x] Automated tests to 90% coverage
- [x] Regression test suite performed for the release along with QA day for all supported OS in https://www.elastic.co/support/matrix 
   - [x] https://github.com/elastic/connectors-python/issues/469
   - [x] MS SQL

~- [ ] Test - ‘native’ version~ not applicable for 8.7

**Comment by parth-elastic** [comment]
@danajuratoni @DianaJourdan We have verified all the checklist for all the dialects. Also the code is backported to 8.7 branch. 

We can close this issue if all looks god to you.

**Comment by seanstory** [comment]
I'm seeing a few open PRs that might relate to this issue:
- https://github.com/elastic/connectors-python/pull/678
- https://github.com/elastic/connectors-python/pull/677
- https://github.com/elastic/connectors-python/pull/627

Do these need to be merged first before we close this issue?

Also, I reviewed the perf8 reports for these 3 connectors (see [this build](https://buildkite.com/elastic/connectors-python-nightly/builds/461)). Largely, things look good. Though I did notice that the Postgres connector's memory consumption is nearly twice what it is for MSSQL and Oracle (400MB vs 200MB). @danajuratoni is that something you'd like to see addressed before we close this, or are you ok with that being a follow-up enhancement?

**Comment by danajuratoni** [comment]
> Postgres connector's memory consumption is nearly twice what it is for MSSQL and Oracle (400MB vs 200MB)

The first threshold is to ingest ~250k docs (or ~10GB) in less than 30 minutes. This has been achieved, so we'll take a look at enhancements as soon as the initial version of the connector is completed.

**Comment by parth-elastic** [comment]
These PRs were dependent on @tarekziade 's PR https://github.com/elastic/connectors-python/pull/624.
However we had a word with @artem-shelkovnikov on Friday and we will be raising a new PR for MS-SQL with pytds library.

We will need to test MS-SQL connector once before raising a PR. So we can keep this issue open untill these PRs gets merged.

**Comment by parth-elastic** [comment]
Hey @artem-shelkovnikov @vidok, All the PRs are merged for Generic Database Connector.  
Just the following point is remaining.

- Sign off given from an engineer in Ingestion for PostgreSQL, Oracle, MSSQL

Please verify it from your side so we can close this issue. 

cc: @nirav-elastic @danajuratoni @DianaJourdan

**Comment by vidok** [comment]
signing off on behalf of the ingestion engineer.

**Comment by parth-elastic** [comment]
@danajuratoni @DianaJourdan I have ticked the signoff by ingestion-engineer tick box hence closing this issue as all the things are completed. feel free to reopen if we have missed any point.
